File: n-th-tribonacci-number/n-th-tribonacci-number.cpp
class Solution {
public:
    int tribs[40];
    int tribonacci(int n) {
        if(n == 0)
            return 0;
        if(n == 1)
            return 1;
        if(n == 2)
            return 1;
        
        if(tribs[n] > 0)
            return tribs[n];
        
        return tribs[n] = tribonacci(n-1) + tribonacci(n-2) + tribonacci(n-3);
    }
};

File: n-th-tribonacci-number/n-th-tribonacci-number.py
class Solution:
    def __init__(self):
        self.dp = [ -1 for i in range(40)]
    def tribonacci(self, n: int) -> int:
        if n == 0:
            return 0
        if n == 1 or n == 2:
            return 1
        
        if self.dp[n] != -1:
            return self.dp[n]
        
        self.dp[n] = self.tribonacci(n - 3) + self.tribonacci(n-2) + self.tribonacci(n-1)
        return self.dp[n]